source/fundamentals/crud/read-operations/cursor.txt

```diff
@@ -1,8 +1,145 @@
+.. _cursor-golang:
+
 =========================
 Access Data From a Cursor
 =========================
 
 .. default-domain:: mongodb
 
-Read operations that return multiple documents do not immediately return
-all values matching the query.
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn how to access data with a **cursor**.
+
+A cursor is a mechanism that allows an application to iterate over
+database results while holding only a subset of them in memory at a
+given time. Read operations that match multiple documents use a cursor
+to return those documents in batches as opposed to all at once.
+
+Sample Cursor
+~~~~~~~~~~~~~
+
+Each section uses the following ``cursor`` variable, which is a
+``Cursor`` struct that contains all the documents in a collection:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CRUD/cursor.go
+   :language: go
+   :dedent:
+   :start-after: begin find
+   :end-before: end find
+
+.. important::
+
+   A cursor is not goroutine safe. Do not use the same cursor in
+   multiple goroutines at the same time.
+
+.. _individual-documents-golang:
+
+Retrieve Documents Individually
+-------------------------------
+
+To retrieve documents from your cursor individually while blocking the
+current goroutine, use the ``Next()`` method.
+
+The method returns a document if each of the following is true:
+
+- A document is currently or will later be available
+- No errors occurred
+- The context didn't expire
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CRUD/cursor.go
+   :language: go
+   :dedent:
+   :start-after: begin cursor next
+   :end-before: end cursor next
+
+Tailable Cursor
+~~~~~~~~~~~~~~~
+
+To attempt retrieving a document from a :manual:`tailable cursor
+</core/tailable-cursors/>`, use the ``TryNext()`` method.
+
+The method returns a document if each of the following is true:
+
+- A document is currently or will later be available
```

Comment:
Optional: `TryNext()` doesn't wait for a document to be available before returning. Update to
> A document is currently available